music21/musicxml/xmlToM21.py

```diff
@@ -2630,18 +2630,19 @@ def xmlForward(self, mxObj: ET.Element):
         if durationText := strippedText(mxDuration):
             change = opFrac(float(durationText) / self.divisions)
 
-            # Create hidden rest (in other words, a spacer)
-            # old Finale documents close incomplete final measures with <forward>
-            # this will be removed afterward by removeEndForwardRest()
-            r = note.Rest(quarterLength=change)
-            r.style.hideObjectOnPrint = True
-            self.addToStaffReference(mxObj, r)
-            self.insertInMeasureOrVoice(mxObj, r)
+            if self.parent.parent.wasWrittenByFinale:
```

Comment:
We need a test that the Finale behavior was preserved. There was one on #1636 you could include and give @TimFelixBeyer co-author credit (which we should do in any case!)

Reply:
Argh.  I actually picked up that test in the next PR up (before I split this PR out from that one).  Sigh.  I'll bring it here.

Reply:
Done.